vm_setup.pl

```diff
@@ -137,6 +141,10 @@
                      "HOSTNAME=$hostname\n";
 close ($etc_network);
 
+# build cpnat before attempting to pull IP address info
+print "building cpnat\n";
+system_formatted("/usr/local/cpanel/scripts/build_cpnat");
```

Comment:
You'd be better off changing #! to /usr/local/cpanel/3rdparty/bin/perl

Then use Cpanel::NAT ();

and do your thing. I've not yet determined what's being done with $ip

Reply:
It looks like I needed a better comment here.  For some reason, openstack does not always populate '/var/cpanel/cpnat' with the correct IPs.  I'm running build_cpnat here in order to ensure that this file has the correct IP information in it as the one-liners (designed to provide a quick access URL to WHM/cPanel/webmail using the create_user_session API call) that the script adds to '/etc/motd' depend on this file being accurate in order for them to work correctly

Reply:
Todd is referring to this:
```
 if (-e("/var/cpanel/cpnat")) { 
    chomp ( $ip = qx(cat /var/cpanel/cpnat | awk '{print\$2}') );
    chomp ( $natip = qx(cat /var/cpanel/cpnat | awk '{print\$1}') );
}
```
The most recent changes to motd generation appear to remove any need for `$ip` and `$natip` so this code block and the respective `my( $ip, $natip )` allocations can be removed.  If this information is needed in the future it would be better to use Cpanel::NAT instead of shelling out to cat/awk.